capture/parse.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import gzip
+import pysam
+import logging
+
+
+from Bio import SeqIO
+from capture import bam
+
+
+def is_gzip(file):
+    logger = logging.getLogger(__name__)
+    magic_number = b"\x1f\x8b\x08\x08"
+    f = open(file, "rb")
+    with f:
+        try:
+            assert f.read(4) == magic_number
+        except AssertionError as e:
+            logger.info(f"{file} is not gzipped")
+            return False
+        else:
+            return True
+
+
+def count_record(file):
+    if is_gzip(file):
+        with gzip.open(file, "rt") as handle:
+            file_record = SeqIO.parse(handle, "fastq")
+            tot_records = sum(1 for line in file_record)
+            return(tot_records)
+    else:
+        with open(file, "rt") as handle:
+            file_record = SeqIO.parse(handle, "fastq")
+            tot_records = sum(1 for line in file_record)
+            return(tot_records)
+
+
+def parse(args, file, type_f, num_sub, number_records):
```

Comment:
could be split  in two functions. parse_bam and parse_fastq

Reply:
same `args` remark than for the split function too. That also why you want two function. The `if` logic to execute `parse_bam` or `parse_fastq` will be in `assemble`